Adds Disqus comments for activities

cjford opened the pull request:
Closes [#156](https://github.com/codefordenver/owlet/issues/156). Any revisions to my nooby clojurescript are welcome.

cjford committed:
Adds Disqus comments for activities (#156)

eemshi reviewed (commented):
@cjford this looks awesome!
but the activity links in the "Latest Discussions" panel are broken - there's a "#!" in the URL when it should just be "#"

![screen shot 2017-02-10 at 11 32 37 pm](https://cloud.githubusercontent.com/assets/10643091/22852216/b0cdc7c4-eff4-11e6-947d-e20ec419797a.png)

eemshi commented on src/cljs/owlet_ui/components/activity/comments.cljs:
```diff
@@ -0,0 +1,15 @@
+(defn load-comments [activity]
+  (js/disqusReset
+    (get-in activity [:sys :id])
+    (clojure.string/replace (-> js/document .-location .-href) "#" "#!")
```
FYI, i tried changing line 8 in my local version to just `(-> js/document .-location .-href)` and then left a comment on [Bug Life](http://localhost:4000/#/activity/2iLyFzilTGsOGEEGWS2gAU) to see if it would show up in "Latest Discussions" with the correct activity link. it hasn't shown up yet

eemshi commented:
the "owlet" title link at the top navigates to http://www.owlet.com.au/ lolz

![screen shot 2017-02-11 at 12 50 23 am](https://cloud.githubusercontent.com/assets/10643091/22852205/6408323a-eff4-11e6-8be4-cbb25a59114e.png)

dviramontes commented on src/cljs/owlet_ui/components/activity/comments.cljs:
```diff
@@ -0,0 +1,15 @@
+    (clojure.string/replace (-> js/document .-location .-href) "#" "#!")
+    (get-in activity [:fields :title])))
+
+(defn activity-comments [activity-info]
```
Thanks @cjford this is great progress!
Can we get rid of this function parameter `activity-info` since its not being used in the function ?

dviramontes commented on resources/public/js/vendor/disqus.js:
```diff
@@ -0,0 +1,23 @@
+var disqus_shortname = 'owlet';
```
i think this line needs to be 
`var disqus_shortname = 'owlet-2';`
based on this: https://owlet-2.disqus.com/admin/settings/universalcode/

cjford commented:
@dviramontes Sounds good, I will update those.

@eemshi Ah so I thought those page URLs were only used as identifiers and didn't realize the links actually show up anywhere. The reason I substituted "#" for "#!" is because it seems to be a [requirement for Disqus](https://help.disqus.com/customer/portal/articles/472107-using-disqus-on-ajax-sites) to detect different pages. In my testing I wasn't able to get separate threads to work using the actual URL with a "#". I don't suppose it's plausible to change the routing scheme or add alternate routes? I will keep looking for another workaround.

eemshi commented:
@cjford oh interesting. maybe we would switch to hash-bang routes... @dviramontes thoughts?

dviramontes commented:
Hi, yeah no strong preference here. We should be to change it to #! The
prefix is set in routes.cljs. let's try it! Thanks team :)

On Mon, Feb 13, 2017, 8:57 PM Michelle L <notifications@github.com> wrote:

> @cjford <https://github.com/cjford> oh interesting. maybe we would switch
> to hash-bang routes... @dviramontes <https://github.com/dviramontes>
> thoughts?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/codefordenver/owlet-ui/pull/95#issuecomment-279584228>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIzYtdClECkfO2QetJ-3D8C2HwwtM3Dks5rcQoCgaJpZM4L96YK>
> .
>

eemshi commented:
~~do you think theres any benefit to changing all the routes (consistency?) or just the activity route (fewer changes?) @cjford @dviramontes~~

EDIT: nvm, i have to change it for all of them.

dviramontes commented:
@eemshi yeah, unfortunately is a global. That being said, we should be able to form our own route pattern like this (as long as we follow this pattern when we link to the activities) 
`/activity/(str "#! activity-id)`

cjford commented:
Updated the original commit:
* Changed disqus_shortname from "owlet" to "owlet-2"
* Removed unused activity-info parameter
* Removed hash to hash-bang URL substitution

Let me know if I can do anything to help with the route prefix changes.

eemshi commented:
hey @cjford, don't we need the hash-bang? or did you find a way around it? i was going to change the routes to include a hash-bang and open a PR against yours. let me know.

cjford commented:
@eemshi Originally I took the real URL (with just the hash) and artificially stuck the hash-bang in before passing it to Disqus [here](https://github.com/codefordenver/owlet-ui/pull/95/files#diff-ac2afb564b75537dc16cd5470f7ccd0aR18). This tricked it into working, but also made all the links wrong. 

If we change the routes to actually use hash-bangs, we can use the real URL without making the substitution. So this PR is now assuming that the routing will have been changed before it gets merged. Does that make sense?

eemshi commented:
@cjford yes my bad, makes sense

On Tue, Feb 14, 2017 at 6:28 PM Connor Ford <notifications@github.com>
wrote:

> @eemshi <https://github.com/eemshi> Originally I took the real URL (with
> just the hash) and artificially stuck the hash-bang in before passing it to
> Disqus here
> <https://github.com/codefordenver/owlet-ui/pull/95/files#diff-ac2afb564b75537dc16cd5470f7ccd0aR18>.
> This tricked it into working, but also made all the links wrong.
>
> If we change the routes to actually use hash-bangs, we can use the real
> URL without making the substitution. So this PR is now assuming that the
> routing will have been changed before it gets merged. Does that make sense?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/codefordenver/owlet-ui/pull/95#issuecomment-279890929>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKJmk6rCQA_wXje6eBfrjUAGjlrD_yZGks5rclSygaJpZM4L96YK>
> .
>

eemshi commented:
@cjford yes my bad, i misunderstood

dviramontes commented:
@cjford 
one thing we can look into in the meantime is this error:
<img width="465" alt="screen shot 2017-02-21 at 7 09 09 pm" src="https://cloud.githubusercontent.com/assets/144226/23195922/717c7da4-f873-11e6-91fb-fb50cedfe3a9.png">
Looks like disqus has trouble bootstrapping itself onto the page.

ZadenRB commented:
@dviramontes @cjford The appendChild issue has been fixed on the [this branch](https://github.com/codefordenver/owlet-ui/tree/feat-156-append-error)

eemshi committed:
Adds #! prefix to activity routes (#104)

* Fix 185 (#96)

* Adds search bar component & styles it, removes breadcrumb component

* feat: init [typeahead] component with opts

* feat: concat skills with branch data

* feat: adds change handler

* Changes styling of search suggestions to css, fixes bug causing suggestion to be registered as selected 3 times as opposed to once

* fix: change change-on-blur?, inline fn for render-suggestion

* amend: debounce delay, feels faster

* refactor: search-bar component include from main-view only, more DRY

* feat: filter activities by branch

* feat: position fix search-bar component

* refactor: result-formatter fn

* fix: suggestion highlight color

* feat: adds helpers ns

* wip: filter by skill

* Fixes styling causing white space next to search bar

* Fixes positioning of search bar with lpsidebar, makes back button a component, fixes back button after performing a search

* feat: filter by skill

* feat: filter by activity title

* fix: return unmodified activity, otherwise it leaves activities out

* Adds function to clear search bar on navigation

* Re-adds clearSearch function to navigation event and updates to remove focus from search bar when cleared

* moves showdown constructor into helpers.cljs (#97)

* FEAT-187: Make search result route work (#99)

* Adds search result route

* Adds search-results view and makes it active view for search results route

* fix: adds clojur.spec check-and-throw fn to helpers/keywordize-name

* Changes clearSearch function into component, routes WIP

* dispatch :filter-activities-by-search-term from :get-activity-metadata-successful when :search route-param exists (#98)

* Makes URL navigate to search results route on :filter-activities-by-search-term

* Calls :set-active-document-title on search, setting it to the search term

* FIX-186: Make search suggestions disappear when search bar loses focus and/or when nothing is typed in search bar (#100)

* wip: fix suggestion count

* Adds functions to hide/show suggestions based on search bar having focus, removes suggestions when nothing is typed

* Refactors hide and show functions into one(WIP)

* Fixes refactored toggle-suggestions function

* Stops printing suggestion-count

* Removes printing of state in toggle-suggestionsfunction

* Adds Disqus comments for activities (#156)

* adds accountant dep, rm route prefix, adds hash-bang to activity route

* adds slashes to relative links

* adds slashes to relative links 2

* fix: dev-handler, catch all on index route

* fix: dev-handler, catch all on /activity/:id

* wip: query params

* Updates routes with #! for activities and # prefix for everything else

* rm accountant dep, uncomments disqus code

* fix: revert dev-handler

* restores search route

* Adds URI decoder to redirect to proper route when links are opened from an external application

* Changes URI decoder to use decodeURIComponent function

* Fixes URI decoder

eemshi committed:
switches to new disqus organization, touches styles

dviramontes committed:
fix: appendChild of null error

eemshi committed:
Merge remote-tracking branch 'origin/master' into feat-156

dviramontes committed:
fix: appendChild of null error 2

eemshi committed:
fix: ()
